jobs/kubelet/templates/bin/drain.erb

```diff
@@ -11,6 +11,11 @@ PIDFILE=/var/vcap/sys/run/kubernetes/kubelet.pid
 exit_code=1
 trap ensure_safe_exit EXIT
 
+node_name="<%= spec.ip %>"
+<% if p("cloud-provider.type") == "gce" %>
+  node_name="$(curl http://metadata.google.internal/computeMetadata/v1/instance/name -H "Bearer: $(curl "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token" -H "Metadata-Flavor: Google" | jq '.access_token')" -H "Metadata-Flavor: Google").c.<%= p("cloud-provider.gce.project-id") %>.internal"
```

Comment:
the token stuff shouldn't be necessary if you're using the default service account. The header is for sure. The following "works on my machine"
```
jrjohnson@omg-ops-manager:~$ curl -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/name"
omg-ops-manager
```